clash-prelude/src/Clash/Explicit/Signal/Delayed.hs

```diff
@@ -49,7 +49,11 @@ import Prelude                    ((.), (<$>), (<*>), id, Num(..))
 import Data.Coerce                (coerce)
 import Data.Kind                  (Type)
 import Data.Proxy                 (Proxy (..))
+#if MIN_VERSION_singletons(3,0,0)
+import Data.Singletons            (Apply, TyFun, type (@@))
+#else
 import Data.Singletons.Prelude    (Apply, TyFun, type (@@))
+#endif
```

Comment:
I checked: this can be `import Data.Singletons` for both singletons 2 and 3. No need for CPP!